MiniSim/Menu.swift

```diff
@@ -121,103 +155,125 @@ class Menu: NSMenu {
+            self.updateSection(with: menuItems, section: section)
         }
     }
     
-    private func populateAndroidSubMenu(booted: Bool) -> NSMenu {
-        let subMenu = NSMenu()
-        for item in AndroidSubMenuItem.allCases {
-            if item == AndroidSubMenuItem.customCommand {
-                continue
-            }
-            
-            let menuItem = item.menuItem
-            menuItem.target = self
-            menuItem.action = #selector(androidSubMenuClick)
-            if item.needBootedDevice && !booted {
-                continue
-            }
-            if item.bootsDevice && booted {
-                continue
-            }
-            subMenu.addItem(menuItem)
+    var sections: [DeviceListSection] {
+        var sections: [DeviceListSection] = []
+        if UserDefaults.standard.enableAndroidEmulators {
+            sections.append(.android)
         }
-        
-        for item in DeviceService.getCustomCommands(platform: .android) {
-            let menuItem = AndroidSubMenuItem.customCommand.menuItem
-            menuItem.target = self
-            menuItem.action = #selector(androidSubMenuClick)
-            if item.needBootedDevice && !booted {
-                continue
-            }
-            if item.bootsDevice ?? false && booted {
-                continue
-            }
-            menuItem.image = NSImage(systemSymbolName: item.icon, accessibilityDescription: item.name)
-            menuItem.title = item.name
-            subMenu.addItem(menuItem)
+        if UserDefaults.standard.enableiOSSimulators {
+            sections.append(.iOS)
         }
-        return subMenu
+        return sections
     }
     
-    private func populateIOSSubMenu(booted: Bool) -> NSMenu {
-        let subMenu = NSMenu()
-        for item in IOSSubMenuItem.allCases {
-            if item == IOSSubMenuItem.customCommand {
-                continue
+    private func filter(devices: [Device], for section: DeviceListSection) -> [Device] {
+        let platform: Platform = section == .iOS ? .ios : .android
+        return devices.filter { $0.platform == platform }
+    }
+    
+    private func updateSection(with items: [NSMenuItem], section: DeviceListSection) {
+        guard let header = self.items.first(where: { $0.tag == section.rawValue }),
+              let startIndex = self.items.firstIndex(of: header) else {
+            return
+        }
+        
+        for (index, menuItem) in items.enumerated() {
+            if let itemIndex = self.items.firstIndex(where: { $0.title == menuItem.title }) {
+                self.replaceMenuItem(at: itemIndex, with: menuItem)
+                return
             }
-            let menuItem = item.menuItem
-            menuItem.target = self
-            menuItem.action = #selector(IOSSubMenuClick)
-            subMenu.addItem(menuItem)
+            self.safeInsertItem(menuItem, at: startIndex + index + 1)
```

Comment:
Something is off with this logic. For me not all items are appended to the menu and also if you create new simulator on the fly (by going to Xcode > Devices & Simualtors and add a new one) its not reflected in the menu after next open. 

Im trying to debug what's wrong there

Reply:
Looks like we should have `continue` instead of `return`

Reply:
Oh, right! It was `forEach` enumeration before.